mutiny-core/src/nostr/nwc.rs

```diff
@@ -166,99 +185,123 @@ impl NostrWalletConnect {
                     node_manager.logger,
                     "NWC Invoice amount not set, cannot pay: {invoice}"
                 );
-                return Ok(None);
+                return Ok((None, needs_save));
             }
 
             // if we have already paid this invoice, skip it
             let node = node_manager.get_node(from_node).await?;
             if node.get_invoice(&invoice).is_ok_and(|i| i.paid) {
-                return Ok(None);
+                return Ok((None, needs_save));
             }
             drop(node);
 
+            if invoice.amount_milli_satoshis().is_none() {
+                log_error!(
+                    node_manager.logger,
+                    "NWC Invoice amount not set, cannot pay: {invoice}"
+                );
+                return Ok((None, needs_save));
+            }
+
             // if we need approval, just save in the db for later
-            if self.profile.require_approval {
-                let pending = PendingNwcInvoice {
-                    index: self.profile.index,
-                    invoice,
-                    event_id: event.id,
-                    pubkey: event.pubkey,
-                };
-                pending_nwc_lock.lock().await;
-
-                let mut current: Vec<PendingNwcInvoice> = node_manager
-                    .storage
-                    .get_data(PENDING_NWC_EVENTS_KEY)?
-                    .unwrap_or_default();
-
-                current.push(pending);
-                current.sort();
-                current.dedup();
-
-                node_manager
-                    .storage
-                    .set_data(PENDING_NWC_EVENTS_KEY, current, None)?;
-
-                return Ok(None);
-            } else {
-                let msats = invoice.amount_milli_satoshis().unwrap();
-
-                // verify amount is under our limit
-                let content = if msats <= self.profile.max_single_amt_sats * 1_000 {
-                    match self
-                        .pay_nwc_invoice(node_manager, from_node, &invoice)
-                        .await
-                    {
-                        Ok(resp) => resp,
-                        Err(e) => Response {
+            match self.profile.spending_conditions.clone() {
+                SpendingConditions::SingleUse(mut single_use) => {
+                    // check if we have already spent
+                    if single_use.spent {
+                        return Ok((None, needs_save));
+                    }
```

Comment:
how do we make sure this is atomic? Should we introduce a lock here? Wouldn't want multiple requests coming in trying to activate this in a race. We already a duplicate problem.

Reply:
We only handle 1 nwc request at a time so it should be safe, but yeah, maybe not the safest to rely on that

Reply:
That's different than paying/approving them right? Because I think we can pay many at the same time. Probably different flow not related to this but yeah maybe just being sure would be good.

Reply:
Yeah, that is separate